features/netsocket/UDPSocket.cpp

```diff
@@ -20,20 +20,32 @@
 
 UDPSocket::UDPSocket()
 {
+    _socket_stats.stats_new_socket_entry(this);
 }
 
 UDPSocket::~UDPSocket()
 {
+    _socket_stats.stats_update_socket_state(this, SOCK_CLOSED);
+}
+
+nsapi_error_t UDPSocket::close()
+{
+    _socket_stats.stats_update_socket_state(this, SOCK_CLOSED);
+    return InternetSocket::close();
```

Comment:
Same is in the `TCPSocket` case. Move the call into `InternetSocket` so we don't need these protocol specific `::close()` methods.